docs/CrossCompilationModel.md

```diff
@@ -0,0 +1,76 @@
+
+# Crosss-Compilation Model
+
+## Components
+
+When compiling Swift code, the compiler will consult three different sources of
+inputs outside of the user provided code.
+
+1. Compiler Resources
+2. System (C/C++) Headers
+3. Platform Swift Modules (SDK Overlay)
+
+These pieces compose in a particular manner to build a system image to build
+code against.
+
+The compiler resources are content for the compiler that the compiler provides.
+In the case of the Swift compiler, this includes the Swift shims. Whilst this is
+a compiler resource, the packaging may not be necessarily part of the toolchain
+due to interdependencies. Some of this content may be required to process the
+system headers themselves (e.g. clang's builtin resource headers).
+
+The C/C++ system headers (and libraries) are what is traditionally called the
+Unix "sysroot". On Darwin, this is the SDK which are shipped to the user, while
+on Windows, this is called the Windows SDK, which is a separate installable
+component.
+
+The platform Swift modules provided by the SDK overlay overlays the sysroot or
+(C/C++) SDK. This overlay is a set of "extensions" to the system provided
+headers allowing for a better import into Swift. This may be in the form of API
+Notes, module maps, wrapper types, or Swift `extension`s. This code may or may
+not be fully inlined into the client code and thus be part of the platform ABI.
+
+## Flags
+
+The compiler resources are controlled via the frontend flag `-resource-dir`.
+This allows the driver to select the correct location in most cases while
+allowing the developer control to override the value if required. This should
+normally not be required to be altered as it is intrinsic to the compiler.
+
+The system headers are more interesting. Since this is C/C++ content, this is
+actually consumed through the clang importer rather than the Swift compiler. The
+Swift toolchain uses clang as the C/C++ compiler on all platforms as it is
+embedded to generate inline FFI to enable seamless C/C++ bridging to Swift. The
+flag used by clang is derived from the GCC toolchain, and is spelt `--sysroot`.
+Again, it is the responsibility of the Swift driver to identify the correct
```

Comment:
If this isn't a run-on sentence, it's a very long one.

perhaps something more like this? Feel free to rework it a bit because I'm still not 100% happy with it.
```suggestion
The compiler driver is responsible for identifying the structure of the sysroot. When cross-compiling, there isn't a consistent location for these files, so the driver must expose an argument to specify where to find these files.
```